libexec/pyenv-exec

```diff
@@ -41,7 +41,20 @@ for script in "${scripts[@]}"; do
 done
 
 shift 1
-# CPython's `sys.executable` requires the `PYENV_BIN_PATH` to be at the top of the `PATH`.
-# https://github.com/pyenv/pyenv/issues/98
-export PATH="${PYENV_BIN_PATH}:${PATH}"
+
+# reset path to original path
+# https://github.com/pyenv/pyenv/issues/985
+# if installing package via pip/conda/easy_install, do not reset path as pyenv-which will be needed for auto rehash
```

Comment:
> pyenv-which will be needed for auto rehash

This should be handled in a better way, e.g. by exporting/setting the location for it explicitly or something similar.

Reply:
Okay, so looking at the auto rehash more carefully, it seems like it needs access to `pyenv-which` and `pyenv-rehash`. I was trying to keep modifications localized in my initial PR.

I can instead, where possible, make `pyenv-xxx` into an absolute reference since these are pre-packaged plugins so I know exactly where they are. I can export the path to the main pyenv plugin directory and re-use it in the plugin (or get the path to the main pyenv plugin directory again within the plugin itself). Does this sound like a better approach to you?

It seems as though the PATH changing is primarily due to the flexibility of running `pyenv-xyz` arbitrarily in plugins.

Reply:
Cannot `PYENV_ROOT` be used to build the absolute path?